### PR TITLE
Mention SDK Modules BOM at various places

### PR DESCRIPTION
**Pull request by Emdee89**
## What Has Changed?

Mentioning the SDK Modules BOM in the docu at various places.

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- [x] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.

### CODE REVIEW

File: docs/java/features/connectivity/destinations.mdx [hunk 1]
```diff
@@ -434,36 +434,36 @@ Furthermore, add the dependency `com.sap.cds:cds-integration-cloud-sdk` to your
...
 For certain services (only BTP Workflow service for now), we provide further convenience that skips the need to provide the `mapping` explicitly.
-  Example:
+Example:
 
-  ```java
-  final HttpDestination destination =
-        ScpCfServiceDestinationLoader.getDestinationForService(
-          ScpCfServiceDestinationLoader.CfServices.WORKFLOW,
-          "my-workflow");
+````java
```
Comment (Johannes-Schneider) [inline docs/java/features/connectivity/destinations.mdx:462]: Is the diff view odd, or are there actually four backticks? If so, please change that back to three instead.

File: docs/java/getting-started.mdx [hunk 2]
```diff
@@ -307,8 +311,9 @@ Check the logs during application startup to ensure the listeners got registered
 </TabItem>
 </Tabs>
 
-:::note
-Please note for Spring based projects we provide out-of-the-box support only on Cloud Foundry with archetype `scp-cf-spring`.
+:::note SAP Cloud SDK Modules Bill-of-Material
+By default, the SAP Cloud SDK archetypes reference the SAP Cloud SDK Bill-of-Material `sdk-bom` which manages dependency versions for all SAP Cloud SDK modules and transitive dependencies.
```
Comment (Johannes-Schneider) [inline docs/java/getting-started.mdx:315]:
```suggestion
By default, the SAP Cloud SDK archetypes reference the SAP Cloud SDK Bill-of-Material `sdk-bom`, which manages dependency versions for all SAP Cloud SDK modules and transitive dependencies.
```

File: docs/java/guides/cap-sdk-integration.mdx [hunk 3]
```diff
@@ -53,6 +53,11 @@ We recommend using the **latest** SAP Cloud SDK version always.
 Check out the [release notes](../release-notes-sap-cloud-sdk-for-java) to see the release history.
 :::
 
+:::note SAP Cloud SDK Modules Bill-of-Material
+By default, the SAP Cloud SDK archetypes reference the SAP Cloud SDK Bill-of-Material `sdk-bom` which manages dependency versions for all SAP Cloud SDK modules and transitive dependencies.
```
Comment (Johannes-Schneider) [inline docs/java/guides/cap-sdk-integration.mdx:57]:
```suggestion
By default, the SAP Cloud SDK archetypes reference the SAP Cloud SDK Bill-of-Material `sdk-bom`, which manages dependency versions for all SAP Cloud SDK modules and transitive dependencies.
```

File: docs/java/guides/dependencies.mdx [hunk 4]
```diff
@@ -57,6 +57,11 @@ It helps in various ways:
 
 For these reasons we highly recommend using the `sdk-bom` in your project.
 
+:::note Alternative: SAP Cloud SDK Modules Bill-of-Material
+Besides the SAP Cloud SDK Bill-of-Material `sdk-bom` there is also the SAP Cloud SDK Modules Bill-of-Material `sdk-modules-bom` which manages the version of all SAP Cloud SDK modules only.
```
Comment (Johannes-Schneider) [inline docs/java/guides/dependencies.mdx:61]:
```suggestion
Besides the SAP Cloud SDK Bill-of-Material `sdk-bom` there is also the SAP Cloud SDK Modules Bill-of-Material `sdk-modules-bom`, which manages the version of all SAP Cloud SDK modules only.
```

**Review by Johannes-Schneider — COMMENTED** [review]
I like the changes. Some minor suggestions.

File: docs/java/features/connectivity/destinations.mdx (hunk 1 above)
Comment (Emdee89) [reply docs/java/features/connectivity/destinations.mdx:462]: I ran the NPM prettifier over it, seems it had messed it up. Will correct it

**Review by Johannes-Schneider — APPROVED** [review]
LGTM.

**Pull request merged**